src/app/home-page/widget/entry-boxes/entry-boxes.component.ts

```diff
@@ -0,0 +1,96 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { EntryType } from 'app/shared/enum/entry-type';
+import { EntryUpdateTime, User, UsersService } from 'app/shared/openapi';
+import { UserQuery } from 'app/shared/user/user.query';
+import { debounceTime, takeUntil } from 'rxjs/operators';
+import { WorkflowService } from 'app/shared/state/workflow.service';
+import { WorkflowQuery } from 'app/shared/state/workflow.query';
+import { MyEntriesQuery } from 'app/shared/state/my-entries.query';
+import { SessionQuery } from 'app/shared/session/session.query';
+import { SessionService } from 'app/shared/session/session.service';
+import { MyWorkflowsService } from 'app/myworkflows/myworkflows.service';
+import { RegisterToolService } from 'app/container/register-tool/register-tool.service';
+import { Base } from 'app/shared/base';
+import { Dockstore } from 'app/shared/dockstore.model';
+
+@Component({
+  selector: 'app-entry-boxes',
+  templateUrl: './entry-boxes.component.html',
+  styleUrls: ['./entry-boxes.component.scss'],
+})
+export class EntryBoxesComponent extends Base implements OnInit {
+  Dockstore = Dockstore;
+  @Input() entryType: string;
+  entryTypeLowerCase: string;
+  filterText: string;
+  listOfEntries: Array<EntryUpdateTime> = [];
+  user: User;
+  helpLink: string;
+  allEntriesLink: string;
+  public isLoading = true;
+
+  constructor(
+    private registerToolService: RegisterToolService,
+    protected userQuery: UserQuery,
+    protected workflowService: WorkflowService,
+    protected workflowQuery: WorkflowQuery,
+    protected myWorkflowsService: MyWorkflowsService,
+    protected myEntriesQuery: MyEntriesQuery,
+    protected sessionQuery: SessionQuery,
+    protected sessionService: SessionService,
+    private usersService: UsersService
+  ) {
+    super();
+    this.user = this.userQuery.getValue().user;
+  }
+
+  ngOnInit(): void {
+    this.getMyEntries();
+    this.entryTypeLowerCase = this.entryType.toLowerCase();
+
+    if (this.entryType === 'Workflow') {
+      this.helpLink = Dockstore.DOCUMENTATION_URL + '/getting-started/dockstore-workflows.html';
+      this.allEntriesLink = '/my-workflows/';
+    } else if (this.entryType === 'Tool') {
+      this.helpLink = Dockstore.DOCUMENTATION_URL + '/getting-started/dockstore-tools.html';
+      this.allEntriesLink = '/my-tools/';
+    } else if (this.entryType === 'Service') {
+      this.helpLink = Dockstore.DOCUMENTATION_URL + '/getting-started/getting-started-with-services.html';
+      this.allEntriesLink = '/my-services/';
+    }
+  }
+
+  getMyEntries() {
+    this.usersService
+      .getUserEntries(10, this.filterText)
```

Comment:
It would be nice if we could just fetch entries of a given type, but that will require an API change.

Reply:
Yeah that was a consideration but would need some backend assistance, it would be a good idea to consider for future changes/improvements